suite/tests/client-interface/reg_size_test.dll.c

```diff
@@ -0,0 +1,38 @@
+#include "dr_api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+DR_EXPORT void
+dr_client_main(client_id_t id, int argc, const char *argv[])
+{
+    dr_set_client_name("DynamoRIO Sample Client 'test_reg_size'",
+                       "http://dynamorio.org/issues");
+
+    opnd_size_t res;
+    for (int i = DR_REG_NULL + 1; i < DR_REG_LAST_VALID_ENUM; ++i) {
+        if (i == DR_REG_INVALID)
+            continue;
+        if (get_register_name(i)[0] == '\0') {
```

Comment:
So this is not testing that there's a name: it's trying to skip reserved-and-unused enum values.  I think this is worth a comment as it's not clear why this is here as at first you'd think it's an error to have no name.

Reply:
So there is no test for names and the missing "cntvct_el0" would not have been detected by this test.  It would read off the end of the array and return garbage.  How about checking that the size of the returned string is small and that each character is ascii alphanumeric?

Reply:
I tried to add tests of register names in the last commit by checking the size and contents as you suggested.

Reply:
LGTM